src/eu/janinko/games/backgammon/Stone.java

```diff
@@ -45,32 +66,50 @@ public int hashCode() {
 
 	@Override
 	public boolean equals(Object obj) {
-		if (this == obj)
-			return true;
-		if (obj == null)
-			return false;
-		if (getClass() != obj.getClass())
+		// If the object is null or not a stone, obj is not equivalent to this
+		if (obj == null || !(obj instanceof eu.janinko.games.backgammon.Stone))
 			return false;
+		// Else compare the Stones by color,
+		// if they match the stones are considered equivalent
```

Comment:
Similarly like ``hashCode``, the ``equals`` method is common with common implementation and when there is nothing surprising that should a reader be pointed to I thing it's not necessary to have there comments.